src/comm.cc

```diff
@@ -783,6 +783,8 @@ commConfigureLinger(const int fd, const OnOff enabled)
     l.l_onoff = (enabled == OnOff::on ? 1 : 0);
     l.l_linger = 0; // how long to linger for, in seconds
 
+    fd_table[fd].flags.harshClosureRequested = (l.l_onoff && !l.l_linger); // close(2) sends TCP RST if true
```

Comment:
This is a low-level private function with callers already checking `fd`, so I think it may be OK to avoid checking `fd` value before accessing fd_table here. If you disagree, consider applying this suggestion:

```suggestion
    if (isOpen(fd))
        fd_table[fd].flags.harshClosureRequested = (l.l_onoff && !l.l_linger); // close(2) sends TCP RST if true
```

Reply:
We would need also to protect setsockopt() by the same check, i.e., basically this function should immediately return if isOpen() is false. I would just assert(isOpen(fd)) instead.

Reply:
Ok, let's leave this function as is.